task.go

```diff
@@ -325,7 +325,6 @@ func (t *task) Delete(ctx context.Context, opts ...ProcessDeleteOpts) (*ExitStat
 		return nil, fmt.Errorf("task must be stopped before deletion: %s: %w", status.Status, errdefs.ErrFailedPrecondition)
 	}
 	if t.io != nil {
-		t.io.Close()
```

Comment:
If this is indeed the right fix, perhaps we should add a comment here to replace the line, and to describe that `Close()` should only be called on a successful Delete (as the comment further down describes). Just in case someone either goes looking for the fix from https://github.com/containerd/containerd/commit/55faa5e93d7fdacc7d9b28ee79a0972ec18e3471, or in case someone wants to add it back ("looks like we forget to close here if we return an error").

Alternatively, we could have a `defer` that handles the close (checks the returned error and if `t.io !=  nil`), but such code can also be easy to break.

Reply:
yeah, I'm curious about whether it should *always* happen (but only after attempting the `Task.Delete()`) or only after a successful one such as that comment seems to imply.